vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java

```diff
@@ -1281,15 +1292,35 @@ public Component getListCellRendererComponent(
       // It is incredibly inefficient for GamePieces and is not needed
       // since we overwrite the label text anyway.
       super.getListCellRendererComponent(list, "", index, selected, hasFocus);
+
+      final int lineOffset = definer.isPrototype() ? 1 : 0;
+
       if (value instanceof EditablePiece) {
-        setText(((EditablePiece) value).getDescription());
+        final EditablePiece editableValue = (EditablePiece) value;
+
+        // Prototype? Skip first (dummy) item to avoid outputting a blank line
+        if (lineOffset == 1 || index > 0) {
+          // For Pieces, bump index to 1
+          // Special formatting for Comments
+          if ((editableValue.getType().startsWith(Comment.ID))) {
+            setText(getLineNumber(index + lineOffset) + "<b>/* " + ConfigureTree.noHTML((editableValue.getDescription())) + " */</b></html>");
+          }
+          else {
+            setText(getLineNumber(index + lineOffset) + ConfigureTree.noHTML(editableValue.getDescription()) + "</html>");
+          }
+        }
       }
       else {
+        // FIXME: When does this get used ? Assumed that prototype might have to be skipped here also.
         final String s = value.getClass().getName();
-        setText(s.substring(s.lastIndexOf('.') + 1));
+        setText(getLineNumber(index + lineOffset) + s.substring(s.lastIndexOf('.') + 1) + "</html>");
       }
       return this;
     }
+
+    private static String getLineNumber(int lineNumber) {
+      return "<html><span style=color:#A0A0A0>" + lineNumber + ".</span> ";  // prep line number for output
+    }
```

Comment:
I would not produce the `<html>` tags unbalanced like this. The function name says it produces the line number, so it should do only that. It would be better to have matching open and close tags produced in the same place.

Reply:
I suggest something like this:
```
private String cellText(int lineno, String s) {
  return "<html>" + getLineNumber(lineno) + s + </html>";
}
```
and then
```
setText(cellText(index + lineOffset, "whatever"));
```